core/src/main/java/tc/oc/pgm/filters/EffectFilter.java

```diff
@@ -31,8 +30,7 @@ protected QueryResponse queryPlayer(PlayerQuery query, MatchPlayer player) {
 
       if (effect.getType().equals(base.getType())
           && (!amplifier || (effect.getAmplifier() == base.getAmplifier()))
-          && ((minDuration == -1) || (effect.getDuration() >= minDuration))
-          && ((maxDuration == -1) || (effect.getDuration() <= maxDuration))) {
+          && duration.contains((long) effect.getDuration())) {
```

Comment:
I was under the impression that getDuration was a long, seems like it's an int, so the Range should also be a `Range<Integer>`, and you should do the cast in the parser instead, but it's a somewhat minor issue